backend/code_groups/cg_controllers.js

```diff
@@ -93,9 +95,188 @@ exports.createGroup = async (req, res) => {
       group: newGroup,
     });
   } catch (error) {
-    res.status(500).json({ message: error.message });
     res.status(500).json({
       error: "Failed to create group",
     });
   }
 };
+
+//Get Group Data for Workstation
+exports.getGroupDetails = async (req, res) => {
+  const { groupId } = req.params;
+  const userId = req.user.id;
+  try {
+    const isMember = await prisma.groupMember.findFirst({
+      where: {
+        groupId: parseInt(groupId),
+        userId: userId,
+      },
+    });
+    const isCreator = await prisma.codeGroup.findFirst({
+      where: {
+        id: parseInt(groupId),
+        creatorId: userId,
+      },
+    });
+
+    if (!isMember && !isCreator) {
+      return res.status(403).json({
+        error: "You do not have access to this group",
+      });
+    }
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        creator: true,
+        members: {
+          include: {
+            user: true,
+            addedBy: true,
+          },
+        },
+        files: true,
+      },
+    });
+    if (!group) {
+      return res.status(404).json({
+        error: "group not found",
+      });
+    }
+    res.status(200).json(group);
+  } catch (error) {
+    res.status(500).json({ message: "Error getting group details" });
+  }
+};
+
+//Get Random Image for Code Group
+exports.getGroupImageUrl = async (req, res) => {
+  const accessKey = process.env.IMAGE_API_ACCESS_KEY;
+  const { query, width, height } = req.query;
+
+  try {
+    const response = await axios.get("https://api.unsplash.com/search/photos", {
+      params: {
+        query: query,
+        per_page: 1,
+      },
+      headers: {
+        Authorization: `Client-ID ${accessKey}`,
+      },
+    });
+    if (response.data.results.length > 0) {
+      const image = await response.data.results[0];
+      const imageUrl = `${image.urls.raw}&w=${width}&h=${height}`;
+      res.status(200).json(imageUrl);
+    } else {
+      res.status(404).json({ message: "No result found" });
+    }
+  } catch (error) {
+    res.status(500).json({ message: "Error getting image" });
+  }
+};
+
+//Update Group Details: Group Name, Image and Add More Members
+exports.updateGroupDetails = async (req, res) => {
```

Comment:
This is a pretty long function, I would consider breaking this up. I don't think it's completely necessary, but in the future I wouldn't have functions any longer than this